src/aggregation/agg_result.rs

```diff
@@ -0,0 +1,37 @@
+use std::collections::HashMap;
+
+use crate::DocId;
+
+pub type AggregationResultTree = HashMap<String, AggregationResult>;
+
+pub enum AggregationResult {
+    BucketResult(BucketAggregationResult),
+    MetricResult(MetricResult),
+}
+
+pub enum MetricResult {
+    Average(i64),
+}
+
+pub struct BucketAggregationResult {
+    buckets: Vec<Box<BucketDataEntry>>,
+}
+
+pub enum Key {
+    String(String),
+    ValueF64(f64),
+    Valuei64(i64),
+}
+
+pub enum BucketDataEntry {
+    KeyCount {
```

Comment:
What will be the alternative here?

Reply:
We have different bucket types for different bucket aggregations, but most common is something with a key and doc count, so I just put that there for now.
Alternative would be a trait, but I think this would cause issues when merging buckets